demo/admin/src/products/generator/ProductHighlightForm.cometGen.ts

```diff
@@ -7,12 +7,22 @@ export default defineConfig<GQLProductHighlight>({
     fragmentName: "ProductHighlightFormDetails",
     fields: [
         { type: "text", name: "description" },
+        {
+            type: "asyncSelectFilter",
+            name: "productCategoryType",
+            rootQuery: "productCategoryTypes",
+            loadValueQueryField: "product.category?.type",
```

Comment:
To me it's strange that loadValueQueryField, which defines a GraphQL field, supports the TypeScript optional operator. Alternatively we could always add the optional operator to nested fields in the generated code (we also do this here: https://github.com/vivid-planet/comet/blob/main/packages/admin/admin-generator/src/commands/generate/generateGrid/generateGrid.ts#L548).